bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeDiscoveryService.java

```diff
@@ -0,0 +1,265 @@
+                    sendSocket.setReuseAddress(true);
+                    sendSocket.setBroadcast(true);
+                    sendSocket.setTimeToLive(2);
+                    sendSocket.joinGroup(new InetSocketAddress(broadcastAddress, DISCOVERY_RESPONSE_PORT),
+                            localNetworkInterface);
+                    receiveSocket.setReuseAddress(true);
+                    sendBroadcastToDiscoverThing(sendSocket, receiveSocket, broadcastAddress);
+                } catch (IOException e) {
+                    logger.warn("Discovery with IO exception: {}", e.getMessage());
+                }
+            });
+        } catch (InterruptedException ie) {
+            // don't care
+        } catch (UnknownHostException e) {
+            logger.warn("Discovery with UnknownHostException exception: {}", e.getMessage());
+        } finally {
+            discoveryActive = false;
+        }
+    }
+
+    private void sendBroadcastToDiscoverThing(MulticastSocket sendSocket, MulticastSocket receiveSocket,
+            InetAddress broadcastAddress) throws IOException {
+        byte[] requestData = DISCOVER_REQUEST.getBytes();
+
+        DatagramPacket request = new DatagramPacket(requestData, requestData.length, broadcastAddress, DISCOVERY_PORT);
+
+        try {
+            sendSocket.send(request);
+        } catch (SocketTimeoutException ste) {
+            // done with scanning
+        }
+
+        receiveSocket.setSoTimeout(INTERFACE_TIMEOUT_SEC * MILLIS_PER_SEC);
+        receiveSocket.setReuseAddress(true);
+        do {
+            byte[] rxbuf = new byte[10240];
+            DatagramPacket packet = new DatagramPacket(rxbuf, rxbuf.length);
+            receiveSocket.receive(packet);
+
+            String response = new String(packet.getData()).trim();
+            logger.trace("Govee Device Response: {}", response);
+
+            final Map<String, Object> properties = getDeviceProperties(response);
+            final Object product = properties.get(GoveeBindingConstants.PRODUCT_NAME);
+            final String productName = (product != null) ? product.toString() : "unknown";
+            final Object mac = properties.get(GoveeBindingConstants.MAC_ADDRESS);
+            final String macAddress = (mac != null) ? mac.toString() : "unknown";
+
+            ThingUID thingUid = new ThingUID(GoveeBindingConstants.THING_TYPE_LIGHT, macAddress.replace(":", "_"));
+
+            DiscoveryResultBuilder discoveryResult = DiscoveryResultBuilder.create(thingUid).withProperties(properties)
+                    .withRepresentationProperty(GoveeBindingConstants.MAC_ADDRESS)
+                    .withLabel("Govee " + productName + " " + properties.get(GoveeBindingConstants.DEVICE_TYPE) + " ("
+                            + properties.get(GoveeBindingConstants.IP_ADDRESS) + ")");
+
+            thingDiscovered(discoveryResult.build());
+        } while (true); // left by SocketTimeoutException
```

Comment:
Perhaps for additional safety:
```suggestion
        } while (!Thread.currentThread().isInterrupted()); // left by SocketTimeoutException
```

Reply:
Sounds like a good idea.

Reply:
It seems you missed the "!", that will probably not work well. 😉

Reply:
Thx for catching that. I was about to debug the issue that only one device was found at a time which explains that.